lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1112,6 +1112,41 @@ LogicalResult BitsPrimOp::canonicalize(BitsPrimOp op,
         rewriter, op, innerBits.getInput(), newHi, newLo);
     return success();
   }
+  // bits(mux(sel, a, b), ...) => mux(sel, bits(a, ...), bits(b, ...))
+  if (auto mux = dyn_cast_or_null<MuxPrimOp>(inputOp)) {
```

Comment:
We should probably do a single early null check and then replace all the `dyn_cast_or_null` with just `dyn_cast`.